Lombiq.NodeJs.Extensions/scripts/copy-assets.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @summary A script to copy files around; expects configuration in assets-to-copy.json or package.json (assetsToCopy).
+ * @description This script is a wrapper around the npm package "copyfiles". It reads its configuration from a file in
+ *              the consuming project named "assets-to-copy.json" or the "assetsToCopy" node in package.json.
+ */
+const { access } = require('node:fs/promises');
+const path = require('path');
+const util = require('util');
+/* eslint-disable-next-line import/no-unresolved -- ESLint does not know where to find external modules; ignore. */
+const copyfiles = util.promisify(require('copyfiles'));
+const getAssetsConfig = require('./get-assets-config');
+
+const verbose = true;
+const filePattern = '**/*';
+
+// Change to consuming project's directory.
+process.chdir('..');
+process.chdir('..');
+
+function log(message) {
+    if (verbose) process.stdout.write(message + '\n');
+}
+
+async function copyFilesFromConfig(assetsConfig) {
+    return Promise.all(assetsConfig
+        .map((assetsGroup) => assetsGroup.sources.map((assetSource) => {
+            // Normalize the relative path to the directory to remove trailing slashes and straighten out any anomalies.
+            const directoryToCopy = path.relative('.', assetSource);
+            log(`directoryToCopy = ${directoryToCopy}`);
+
+            return access(directoryToCopy)
+                .catch(() => process.stderr.write(`The directory "${directoryToCopy}" cannot be accessed to copy files from.\n`))
+                .then(() => {
```

Comment:
Merge these two together (instead of `.catch(f1).then(f2)` it should be `.then(f2, f1)`). Why? Because a resolved `.catch` is still a successful promise. So if you can't access the directory it will write out the message and then try to access the directory with the `.then` anyway.

You can test this by typing
```javascript
Promise.reject('expected')
    .catch(() => 'unexpected')
    .then((result) => console.log(result)); 
```
It will print "unexpected".

Using `.then(resolveCallback, rejectCallback)` format avoids this. Alternatively, you can move the `.catch` underneath the `.then`. Then any failure from either the `access()` or the `.then` will be handled by it.

Reply:
Thanks, that made it crystal clear. 😎